### ISSUE TITLE
OSX 10.11.2, Node 5.6.0, abp-filter-parser-cpp@1.1.7 failure

### ISSUE THREAD

**Issue by weems** [brave/browser-laptop#807]
Please see the log at the bottom of this Issue.
```
npm verb stack Error: abp-filter-parser-cpp@1.1.7 install: `node-gyp rebuild`
npm verb stack Exit status 1
npm verb stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/lifecycle.js:232:16)
npm verb stack     at emitTwo (events.js:100:13)
npm verb stack     at EventEmitter.emit (events.js:185:7)
npm verb stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:24:14)
npm verb stack     at emitTwo (events.js:100:13)
npm verb stack     at ChildProcess.emit (events.js:185:7)
npm verb stack     at maybeClose (internal/child_process.js:827:16)
npm verb stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
npm verb pkgid abp-filter-parser-cpp@1.1.7
npm verb cwd /Users/weems/code/brave/browser-laptop
npm ERR! Darwin 15.2.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "--verbose"
npm ERR! node v5.6.0
npm ERR! npm  v3.6.0
npm ERR! code ELIFECYCLE

npm ERR! abp-filter-parser-cpp@1.1.7 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the abp-filter-parser-cpp@1.1.7 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the abp-filter-parser-cpp package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs abp-filter-parser-cpp
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls abp-filter-parser-cpp
npm ERR! There is likely additional logging output above.
npm verb exit [ 1, true ]

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/weems/code/brave/browser-laptop/npm-debug.log
```
Version:  
```
node -v && npm -v
v5.6.0
3.6.0
```
Log: [npm-debug.txt](https://github.com/brave/browser-laptop/files/135158/npm-debug.txt)

**Comment by weems** [comment]
@bbondy building not from npm but from master here (npm install --global inside abp-filter-parser folder) resulted with zero build errors. now will try building again in browser-laptop

**Comment by bbondy** [comment]
what did you install globally? Note that this wouldn't be right to suggest in a wiki.

**Comment by weems** [comment]
@bbondy  

1. git clone --recursive https://github.com/bbondy/abp-filter-parser-cpp.git
2. cd abp-filter-parser-cpp
3. sudo npm install --global
Then it builds successfully:
1. cd ..
2. cd browser-laptop
3. npm install --verbose
4. And I still see the following:

```
npm verb stack Error: abp-filter-parser-cpp@1.1.7 install: `node-gyp rebuild`
npm verb stack Exit status 1
npm verb stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/lifecycle.js:232:16)
npm verb stack     at emitTwo (events.js:100:13)
npm verb stack     at EventEmitter.emit (events.js:185:7)
npm verb stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:24:14)
npm verb stack     at emitTwo (events.js:100:13)
npm verb stack     at ChildProcess.emit (events.js:185:7)
npm verb stack     at maybeClose (internal/child_process.js:827:16)
npm verb stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
npm verb pkgid abp-filter-parser-cpp@1.1.7
npm verb cwd /Users/weems/code/brave/browser-laptop
npm ERR! Darwin 15.2.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "--verbose"
npm ERR! node v5.6.0
npm ERR! npm  v3.6.0
npm ERR! code ELIFECYCLE

npm ERR! abp-filter-parser-cpp@1.1.7 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the abp-filter-parser-cpp@1.1.7 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the abp-filter-parser-cpp package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs abp-filter-parser-cpp
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls abp-filter-parser-cpp
npm ERR! There is likely additional logging output above.
npm verb exit [ 1, true ]

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/weems/code/brave/browser-laptop/npm-debug.log
```
Somehow I need to be able to specify the abp-filter-parser-cpp version I built from git and not from npm.

**Comment by weems** [comment]
@bbondy  Figured this out. Not sure why but the version of abp-filter-parser-cpp@1.1.7 from npm's server could not be installed / built on my machine. The following steps resolved my issue
1. ``` git clone https://github.com/brave/abp-filter-parser-cpp ```
2. ``` cd abp-filter-parser-cpp ```
3. ``` sudo npm link ```
4. ``` cd .. ```
5. ``` cd browser-laptop ```
6. ``` sudo npm link abp-filter-parser-cpp ```
7. ``` sudo npm install ``` which is not able to install successfully.

Since the issue is now resolved, I will close the issue. Not sure if the npm issue is only affecting me for some reason or what the reason was originally, this seems to solve for now.

**Comment by bbondy** [comment]
This is a workaround but it comes down to not the right general solution and you're missing some dependency like maybe the right version of node-gyp

**Comment by weems** [comment]
@bbondy I don't know if it's the latest but:
``` $ node-gyp --version
v3.2.1
```

**Comment by bbondy** [comment]
Those are all the correct error messages.  Ya let's leave this open until we find the right resolution, thanks.

**Comment by luixxiul** [comment]
Installing Node 5.9, instead of 5.6, is the right solution for the moment, isn't it?
See: https://github.com/brave/browser-laptop/issues/244#issuecomment-198834425

**Comment by luixxiul** [comment]
Fixed with #1682

**Comment by bbondy** [comment]
pls try again with updated deps.
